ingestion-beam/src/test/java/com/mozilla/telemetry/DecoderMainTest.java

```diff
@@ -116,4 +116,37 @@ public void testGzippedPayload() throws Exception {
     assertThat("Main output differed from expectation", outputLines,
         matchesInAnyOrder(expectedOutputLines));
   }
+
+  @Test
+  public void testIdempotence() throws Exception {
+    String outputPath = outputFolder.getRoot().getAbsolutePath();
+    String resourceDir = "src/test/resources/testdata/decoder-integration";
+    String input = resourceDir + "/*-input.ndjson";
+    String intermediateOutput = outputPath + "/out1/out1";
+    String output = outputPath + "/out2/out2";
+    String errorOutput = outputPath + "/error/error";
+
+    Decoder.main(new String[] { "--inputFileFormat=json", "--inputType=file", //
+        "--input=" + input, //
+        "--output=" + intermediateOutput, //
+        "--errorOutput=" + errorOutput, //
+        "--outputFileFormat=json", "--outputType=file", "--errorOutputType=file",
+        "--includeStackTrace=false", "--outputFileCompression=UNCOMPRESSED",
+        "--errorOutputFileCompression=UNCOMPRESSED", "--geoCityDatabase=GeoLite2-City.mmdb",
+        "--schemasLocation=schemas.tar.gz", "--redisUri=" + redis.uri });
+
+    Decoder.main(new String[] { "--inputFileFormat=json", "--inputType=file", //
+        "--input=" + intermediateOutput + "*.ndjson", //
+        "--output=" + output, //
+        "--errorOutput=" + errorOutput, //
+        "--outputFileFormat=json", "--outputType=file", "--errorOutputType=file",
+        "--includeStackTrace=false", "--outputFileCompression=UNCOMPRESSED",
+        "--errorOutputFileCompression=UNCOMPRESSED", "--geoCityDatabase=GeoLite2-City.mmdb",
```

Comment:
using `GeoLite2-City.mmdb` is going to conflict with #558, so that will need to be resolved in the pr merged second

Reply:
Indeed